### PR TITLE
Improve candidate submissions review experience

### PR DESCRIPTION
**Pull request by oliviertassinari**
Related to https://app.ashbyhq.com/interview-briefings/7794e311-0a0a-45c6-bc2b-9765417b944c/briefing/right-side/candidates/ca521a45-6b38-40c5-aa65-9cebcf6a55fc/applications/f1566e0d-63ef-4428-b555-11ed70249efa/notes

As a side note, In https://www.notion.so/mui-org/Technical-challenge-b5e93122132f44ce81c6a25b23116ee0, I listed Doctolib, the first company I worked at. They were provided a set of tests to make passes. This might help candidates in our case, give some structure. I also remember the hiring managers were looking if the candidate was adding more tests or not.

### CODE REVIEW

File: pages/api/connection.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { Pool, DatabaseError } from "pg";
 
 const pool = new Pool({
-  connectionString: `postgres://postgres:postgres@localhost:5432/postgres`,
+  connectionString: `postgres://postgres:postgres@localhost:5433/postgres`,
```
Comment (oliviertassinari) [inline pages/api/connection.ts:5]: It conflicts with another local running PostgreSQL instance of my localhost.

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "5.0.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "next dev --port 3002",
```
Comment (oliviertassinari) [inline package.json:6]: It conflicts with my local MUI Core and MUI X running servers, matches https://www.notion.so/mui-org/Ports-b113d0d4c76c4db9a0ef99c703d0084c#a1c936d8f6324c4f9030a1b5be99d54e.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,16 @@
 # Full-stack technical challenge @ MUI
 
-This challenge is part of the hiring process at MUI for the React Engineer position. The idea is to make as much progress as possible under a given time constraint (3-4 hours).
+This challenge is part of the hiring process at MUI for the React Engineer position.
+The idea is to make as much progress as possible under a given time constraint (3-4 hours).
 
 ## Context
 
-MUI is looking for fullstack engineers to join the Toolpad team. We search people who are proficient in both modern react development and backend technologies. Throughout this challenge you are required to build a basic forum application.
+MUI is looking for Full-stack engineers to join the team.
```
Comment (oliviertassinari) [inline README.md:8]: I have removed "Toolpad" to avoid confusion.

File: package.json (hunk 2 above)
Comment (Janpot) [reply package.json:6]: Doesn't next.js choose the next available port on its own?

Comment (oliviertassinari) [reply package.json:6]: It does. The issue for me is that it's not predictable.

Screenshot 2022-11-16 at 18 17 26

https://github.com/mui/fullstack-technical-challenge#run-instructions

Another use case: sometimes I have crashes, say 3001 crashes. I then open a candidate submission, it takes the slot. Later on, I want to check something on MUI X: the port, I open 3001: nop, wrong one.

File: pages/api/connection.ts (hunk 1 above)
Comment (Janpot) [reply pages/api/connection.ts:5]: Might be confusing to candidates who are used to work with the standard 5432 port. ~Perhaps we should just run a public instance with readonly access in render.com?~ nvm, the challenge consists of write use-cases as well

File: package.json (hunk 2 above)
Comment (Janpot) [reply package.json:6]: In all fairness, candidates doing this challenge won't have MUI servers running locally. I mean, what if they have something running on 3002? If the candidate can't easily find their way out of the situation where something is already running on the standard port we propose, I'd consider that part of the challenge and a big red flag.

Perhaps we should word that as "If everything goes well, a message in the console will contain the url of the running application."?

**Review by Janpot — APPROVED** [review]
I don't really have anything against these changes, but personally I think we can consider it part of the challenge that a fullstack engineer can set up this repo with default ports assigned.

_edit: If we add tests, I'd consider adding an integration test as to leave the choices for implementation as open as possible_

File: package.json (hunk 2 above)
Comment (oliviertassinari) [reply package.json:6]: > In all fairness, candidates doing this challenge won't have MUI servers running locally.

Agree but to be clear, these changes are aimed at saving time to the MUI's engineers that will **review** the challenges. My hope is that they are transparent for candidates.

**Comment by oliviertassinari** [comment]
@Janpot regarding my comment about the tests, what's your take on this?

**Pull request merged**